code/controllers/subsystem/mobs.dm

```diff
@@ -26,18 +27,54 @@ SUBSYSTEM_DEF(mobs)
 		dead_players_by_zlevel[dead_players_by_zlevel.len] = list()
 
 /datum/controller/subsystem/mobs/fire(resumed = FALSE)
-	if (!resumed)
-		src.currentrun = GLOB.mob_living_list.Copy()
+	if(!resumed)
+		currentrun = GLOB.mob_living_list.Copy()
 
-	//cache for sanic speed (lists are references anyways)
-	var/list/currentrun = src.currentrun
+	///Cache for speed as lists are references
+	var/list/currentprocrun = currentrun
 	var/times_fired = src.times_fired
-	while(currentrun.len)
-		var/mob/living/L = currentrun[currentrun.len]
-		currentrun.len--
-		if(L)
-			L.Life(times_fired)
-		else
-			GLOB.mob_living_list.Remove(L)
-		if (MC_TICK_CHECK)
+	while(currentprocrun.len)
+		var/mob/living/L = currentprocrun[currentprocrun.len]
+		currentprocrun.len--
+		if(!L.life_process())
+			GLOB.mob_living_list.Remove(L)//we died
```

Comment:
I forsee this causing alot of bugs.

must_call_parent() doesn't help if the proc runtimes before calling parent or forgets to return the right value.

I also think this logic might be a bit disconnected. How likely is a new programmer to our project going to even think that not returning true in life_process() removes the mob from the mob_living_list and stops it from processing. the two events seem unrelated.

Also this is likely fucking up your benchmarks as life() did not do this so you may be having cases where something is runtiming or otherwise not returning TRUE, and that doesn't process on future ticks, but in life() it would process in future ticks. 

Did you maybe mean to check just if L was null?